Detectors/Vertexing/src/SVertexer.cxx

```diff
@@ -477,15 +480,32 @@ void SVertexer::buildT2V(const o2::globaltracking::RecoContainer& recoData) // a
       }
       const auto& trc = recoData.getTrackParam(tvid);
 
+      bool hasTPC = false;
       bool heavyIonisingParticle = false;
+      bool compatibleWithProton = mSVParams->mFractiondEdxforCascBaryons > 0.999f; // if 1 or above, accept all regardless of TPC
       auto tpcGID = recoData.getTPCContributorGID(tvid);
       if (tpcGID.isIndexSet() && isTPCloaded) {
+        hasTPC = true;
         auto& tpcTrack = recoData.getTPCTrack(tpcGID);
         float dEdxTPC = tpcTrack.getdEdx().dEdxTotTPC;
         if (dEdxTPC > mSVParams->minTPCdEdx && trc.getP() > mSVParams->minMomTPCdEdx) // accept high dEdx tracks (He3, He4)
         {
           heavyIonisingParticle = true;
         }
+        auto protonId = o2::track::PID::Proton;
+        float dEdxExpected = mPIDresponse.getExpectedSignal(tpcTrack, protonId);
+        float fracDevProton = std::abs((dEdxTPC - dEdxExpected) / dEdxExpected);
+        if (fracDevProton < mSVParams->mFractiondEdxforCascBaryons) {
+          compatibleWithProton = true;
+        }
+      }
+
+      // get Nclusters in the ITS if available
+      uint8_t nITSclu = -1;
+      auto itsGID = recoData.getITSContributorGID(tvid);
+      if (itsGID.isIndexSet() && isITSloaded) {
```

Comment:
This can be also an ITS-TPC afterburner track, one should change it to
```
if (itsGID.getSource()==GIndex::ITS && isITSloaded) {
```

Reply:
OK, changed now! Thanks!